src/Enforcer.php

```diff
@@ -237,12 +243,11 @@ public function getPermissionsForUser(string $user, string ...$domain): array
         $permission = [];
         foreach ($this->model['p'] as $ptype => $assertion) {
             $args = [];
-            $args[0] = $user;
-            foreach ($assertion->tokens as $i => $token) {
-                if ($token == sprintf('%s_dom', $ptype)) {
-                    $args[$i] = $domain[0];
-                    break;
-                }
+            $subIndex = $this->model->getFieldIndex('p', Constants::SUBJECT_INDEX);
+            $args[$subIndex] = $user;
+            if (\count($domain) > 0) {
+                $domIndex = $this->model->getFieldIndex($ptype, Constants::DOMAIN_INDEX);
+                $args[$domIndex] = $domain[0];
```

Comment:
Not covered by unit tests
https://coveralls.io/builds/68082308/source?filename=src%2FEnforcer.php#L249

Reply:
Okay, actually these have been covered in my second commit in this report [Build 9513259525](https://coveralls.io/builds/68087221). 👻